src/backend/columnar/columnar_storage.c

```diff
@@ -699,9 +700,12 @@ WriteToBlock(Relation rel, BlockNumber blockno, uint32 offset, char *buf,
 			 uint32 len, bool clear)
 {
 	Buffer buffer = ReadBuffer(rel, blockno);
+	GenericXLogState *state = GenericXLogStart(rel);
+
```

Comment:
I'm not sure if it would make any differences but an example pattern from pg codebase (blinsert) seems to be calling `GenericXLogStart` after locking buffer. So maybe we should move this line to L706 ?

Reply:
The [Generic WAL Doc](https://www.postgresql.org/docs/current/generic-wal.html) is pretty clear: "Exclusive lock must be held on each target buffer from before GenericXLogRegisterBuffer() until after GenericXLogFinish()." 

So I think it's fine.